packages/data/src/registry.js

```diff
@@ -213,7 +231,16 @@ export function createRegistry( storeConfigs = {}, parent = null ) {
 		return parent.__experimentalSubscribeStore( storeName, handler );
 	}
 
+	function batch( callback ) {
+		emitter.pause();
+		forEach( stores, ( store ) => store.emitter.pause() );
+		callback();
+		emitter.resume();
+		forEach( stores, ( store ) => store.emitter.resume() );
+	}
+
 	let registry = {
+		batch,
```

Comment:
Should we mark it as `unstable` or `experimental`?

Reply:
I'm personally pretty confident about the semantics and need for this API but can mark experimental if needed. Do you think it's something that is likely to be revisited?

Reply:
If it's something that's stable enough then we should consider adding documentation about it. Comparing to other experimental features we've added, this feels relatively new and not well-known to other developers. (I don't think even I fully understand this feature either.)

The naming of this could also be mistaken to the Redux's `batch` function. I believe there're needs to properly document this and make sure contributors know when and how to use it.

Reply:
The good news is that this is exactly the same thing as react-redux's batch function. Funny because I didn't know about it before your comment :) 

Good point on the docs, I'll add some.

Reply:
Huh, I thought you said that they were different (https://github.com/WordPress/gutenberg/pull/34046#discussion_r688402621)? If it's exactly the same as `react-redux`'s batch function, why don't we just use the same implementation ([`unstable_batchedUpdates`](https://github.com/reduxjs/react-redux/blob/d622cb531a/src/utils/reactBatchedUpdates.ts))? 🤔

Reply:
Because we do a bit more than just `unstable_batchedUpdates` we also avoid calling `subscribe` meaning we avoid two steps: "calling selectors" and "rerendering" while "unstable_batchedUpdates)" avoids just one step "rerendering".

I'd say the goal is the same but the implementation is different in order to have more gains (double)